samples/nrf9160/serial_lte_modem/src/lwm2m/th_lwm2m_service.c

```diff
@@ -0,0 +1,727 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#define LOG_DOMAIN lwm2m
+#define LOG_LEVEL CONFIG_THIN_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#include <zephyr.h>
+#include <gpio.h>
+#include <sensor.h>
+#include <net/lwm2m.h>
+#include "th_connect.h"
+#include "th_util.h"
+#include "th_lwm2m_service.h"
+
+#if defined(CONFIG_THIN_LWM2M_PEER_HOST_IP)
+#define SERVER_ADDR CONFIG_THIN_LWM2M_PEER_HOST_IP
+#else
+#define SERVER_ADDR CONFIG_THIN_LWM2M_PEER_HOST_NAME
+#endif
+
+#define CLIENT_MANUFACTURER	"Zephyr"
+#define CLIENT_MODEL_NUMBER	"OMA-LWM2M Sample Client"
+#define CLIENT_SERIAL_NUMBER	"345000123"
+#define CLIENT_FIRMWARE_VER	"1.0"
+#define CLIENT_DEVICE_TYPE	"OMA-LWM2M Client"
+#define CLIENT_HW_VER		"1.0.1"
+
+#define MAX_URI_LENGTH		16
+#define MAX_READ_LENGTH		255
+
+#define LIGHT_NAME1		"Test light 1"
+#define LIGHT_NAME2		"Test light 2"
+
+enum lwm2m_int_type {
+	LWM2M_INT_TYPE_BOOLEAN,
+	LWM2M_INT_TYPE_UINT8,
+	LWM2M_INT_TYPE_UINT16,
+	LWM2M_INT_TYPE_UINT32,
+	LWM2M_INT_TYPE_UINT64,
+	LWM2M_INT_TYPE_INT8,
+	LWM2M_INT_TYPE_INT16,
+	LWM2M_INT_TYPE_INT32,
+	LWM2M_INT_TYPE_INT64
+};
+
+enum lwm2m_float_type {
+	LWM2M_FLOAD_TYPE_32,
+	LWM2M_FLOAD_TYPE_64
+};
+
+static u8_t bat_idx = LWM2M_DEVICE_PWR_SRC_TYPE_BAT_INT;
+static int bat_mv = 3800;
+static int bat_ma = 125;
+static u8_t usb_idx = LWM2M_DEVICE_PWR_SRC_TYPE_USB;
+static int usb_mv = 5000;
+static int usb_ma = 900;
+static u8_t bat_level = 95;
+static u8_t bat_status = LWM2M_DEVICE_BATTERY_STATUS_CHARGING;
+static int mem_free = 15;
+static int mem_total = 25;
+
+static struct lwm2m_ctx client;
+
+#if defined(CONFIG_LWM2M_DTLS_SUPPORT)
+#define TLS_TAG			1
+
+/* "000102030405060708090a0b0c0d0e0f" */
+static unsigned char client_psk[] = {
+	0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+	0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+};
+
+static const char client_psk_id[] = "Client_identity";
```

Comment:
I wonder if we could be consistent with `lwm2m_client` sample here, and use `nrf-IMEI` format for the PSK ID? So just that both samples can run on the same server with the same credentials? See:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/d390b33b47ba4219edf6ce8fa46a56ab6703f24b/samples/nrf9160/lwm2m_client/src/main.c#L220

Reply:
This could be a TO-DO. This sample project does not address Security in LwM2M and MQTT yet.